host-ext/crouton/background.js

```diff
@@ -404,6 +411,53 @@ function websocketMessage(evt) {
         }
 
         break;
+    case 'N': /* Raise a notification */
+        /* Payload in JSON format, compatible with chrome.extensions specifications */
+        try {
+            var data = JSON.parse(payload);
+
+            if (!data.type)
+                data.type = "basic";
+            if (!data.iconUrl)
+                data.iconUrl = "icon-128.png";
```

Comment:
Can icon data be serialized into the json and then extracted?  Being able to set the icon to a chroot-local file is important for notifications...